pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -82,3 +82,44 @@ func resourceName(name string) string {
 	parts := strings.Split(name, "/")
 	return parts[len(parts)-1]
 }
+
+func (ca *cloudCIDRAllocator) NetworkToNodes(network *networkv1.Network) error {
+	k8sNodesList, err := ca.nodeLister.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("error fetching nodes: %v", err)
+	}
+	// Need to reschedule all Nodes, since we do not know which one holds the
+	// new Network.
+	for _, node := range k8sNodesList {
+		if network != nil {
```

Comment:
if network is nil do we need to do something? can't we just check at the beginning at exit fast?

Reply:
when network==nil we will run ca.AllocateOrOccupyCIDR(node) for every node.